src/gui/AutomationPatternView.cpp

```diff
@@ -313,27 +314,35 @@ void AutomationPatternView::paintEvent( QPaintEvent * )
 		}
 
 		float *values = m_pat->valuesAfter( it.key() );
-		for( int i = it.key(); i < (it + 1).key(); i++ )
+
+		QPainterPath path;
+		QPointF origin = QPointF(x_base + it.key() * ppTact,0.0f);
+		path.moveTo( origin );
+		path.moveTo( QPointF( x_base + it.key() * ppTact,values[0] ) );
+		for( int i = it.key() + 1; i < ( it + 1 ).key(); i++ )
 		{
-			float value = values[i - it.key()];
-			const float x1 = x_base + i * ppt /
-						MidiTime::ticksPerTact();
-			const float x2 = x_base + (i + 1) * ppt /
-						MidiTime::ticksPerTact();
+			const float x1 = x_base + i * ppTact;
```

Comment:
as there is no `x2` now, this variable could be called just `x` to avoid confusion.